libcnb-test/src/test_runner.rs

```diff
@@ -59,9 +59,10 @@ impl TestRunner {
     ) {
         let config = config.borrow();
 
-        let cargo_manifest_dir = env::var("CARGO_MANIFEST_DIR")
-            .map(PathBuf::from)
-            .expect("Could not determine Cargo manifest directory");
+        let cargo_manifest_dir = env::var("CARGO_MANIFEST_DIR").map_or_else(
+            |error| panic!("Error determining Cargo manifest directory: {error}"),
+            PathBuf::from,
+        );
```

Comment:
Then I first read this, it made me stop for a moment and made me think about what's happening here. `.expect` seems a lot more natural.

However, `.expect` doesn't use the display representation which I assume is the reason we changed this to the new form. Can we add a short comment explaining this?

Reply:
It's explained in the PR description

Reply:
I agree it's not immediately obvious about the implications between `.expect()` and a hand-written `panic!` (which is why I covered it in the PR description), but I think it's a bit too far towards "explaining how Rust works" to include an inline comment?

Reply:
Also, this is using the same style already used in other places, such as:
https://github.com/heroku/libcnb.rs/blob/5ab4b036a8dd18dc2d25a3d3ee49ed66ef3ec9d3/libcnb-test/src/test_runner.rs#L111-L127

Reply:
I'll leave the final decision up to you - it's not a big deal. The other example seems even less problematic since only one operation takes place. I think what made me pause for a moment is the `map_or_else` where the else case is a side-effect.

Reply:
The initial edit I made was only `expect` -> `panic!`, however, Clippy then reported:

```
warning: called `map(<f>).unwrap_or_else(<g>)` on a `Result` value. This can be done more directly by calling `.map_or_else(<g>, <f>)` instead
  --> libcnb-test/src/test_runner.rs:62:34
   |
62 |           let cargo_manifest_dir = env::var("CARGO_MANIFEST_DIR")
   |  __________________________________^
63 | |             .map(PathBuf::from)
64 | |             .unwrap_or_else(|error| panic!("Error determining Cargo manifest directory: {error}"));
   | |__________________________________________________________________________________________________^ help: try: `env::var("CARGO_MANIFEST_DIR").map_or_else(|error| panic!("Error determining Cargo manifest directory: {error}"), PathBuf::from)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_unwrap_or
```